--- original/net/minecraft/client/Minecraft.java
+++ changed/net/minecraft/client/Minecraft.java
@@ -35,6 +35,15 @@
 import java.util.concurrent.Executors;
 import java.util.concurrent.FutureTask;
 import javax.imageio.ImageIO;
+
+import dev.minecraftbox.MinecraftBox;
+import dev.minecraftbox.event.EventBus;
+import dev.minecraftbox.event.events.core.InitializationEvent;
+import dev.minecraftbox.event.events.core.PostInitializationEvent;
+import dev.minecraftbox.event.events.core.PreInitializationEvent;
+import dev.minecraftbox.event.events.gui.GuiCloseEvent;
+import dev.minecraftbox.event.events.gui.GuiOpenEvent;
+import dev.minecraftbox.event.events.world.WorldLoadEvent;
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.client.audio.MusicTicker;
@@ -408,6 +417,10 @@
      * Starts the game: initializes the canvas, the title, the settings, etcetera.
      */
     private void startGame() throws LWJGLException, IOException {
+        EventBus.INSTANCE.register(MinecraftBox.INSTANCE);
+        EventBus.INSTANCE.post(new PostInitializationEvent());
+
+        EventBus.INSTANCE.post(new PreInitializationEvent());
         this.gameSettings = new GameSettings(this, this.mcDataDir);
         this.defaultResourcePacks.add(this.mcDefaultResourcePack);
         this.startTimerHackThread();
@@ -524,6 +537,8 @@
         }
 
         this.renderGlobal.makeEntityOutlineShader();
+
+        EventBus.INSTANCE.post(new InitializationEvent());
     }
 
     private void registerMetadataSerializers() {
@@ -857,6 +872,9 @@
         this.currentScreen = (GuiScreen)guiScreenIn;
 
         if (guiScreenIn != null) {
+            GuiOpenEvent guiOpenEvent = new GuiOpenEvent(guiScreenIn);
+            EventBus.INSTANCE.post(guiOpenEvent);
+            if (guiOpenEvent.isCancelled()) return;
             this.setIngameNotInFocus();
             ScaledResolution scaledresolution = new ScaledResolution(this);
             int i = scaledresolution.getScaledWidth();
@@ -864,6 +882,9 @@
             ((GuiScreen)guiScreenIn).setWorldAndResolution(this, i, j);
             this.skipRenderWorld = false;
         } else {
+            GuiCloseEvent guiCloseEvent = new GuiCloseEvent();
+            EventBus.INSTANCE.post(guiCloseEvent);
+            if (guiCloseEvent.isCancelled()) return;
             this.mcSoundHandler.resumeSounds();
             this.setIngameFocus();
         }
@@ -1927,6 +1948,10 @@
      * par2Str is displayed on the loading screen to the user unloads the current world first
      */
     public void loadWorld(WorldClient worldClientIn, String loadingMessage) {
+        WorldLoadEvent worldLoadEvent = new WorldLoadEvent(worldClientIn, loadingMessage);
+        EventBus.INSTANCE.post(worldLoadEvent);
+        if (worldLoadEvent.isCancelled()) return;
+
         if (worldClientIn == null) {
             NetHandlerPlayClient nethandlerplayclient = this.getNetHandler();
 
