--- original/net/minecraft/util/RegistrySimple.java
+++ changed/net/minecraft/util/RegistrySimple.java
@@ -14,7 +14,7 @@
     protected final Map<K, V> registryObjects = this.createUnderlyingMap();
 
     protected Map<K, V> createUnderlyingMap() {
-        return Maps.<K, V>newHashMap();
+        return Maps.newConcurrentMap();
     }
 
     public V getObject(K name) {
@@ -29,14 +29,14 @@
         Validate.notNull(value);
 
         if (this.registryObjects.containsKey(key)) {
-            logger.debug("Adding duplicate key \'" + key + "\' to registry");
+            logger.debug("Adding duplicate key '" + key + "' to registry");
         }
 
         this.registryObjects.put(key, value);
     }
 
     public Set<K> getKeys() {
-        return Collections.<K>unmodifiableSet(this.registryObjects.keySet());
+        return Collections.unmodifiableSet(this.registryObjects.keySet());
     }
 
     /**
