--- original/net/minecraft/init/Blocks.java
+++ changed/net/minecraft/init/Blocks.java
@@ -240,7 +240,7 @@
      * @param blockName The block's name to register
      */
     private static Block getRegisteredBlock(String blockName) {
-        return (Block)Block.blockRegistry.getObject(new ResourceLocation(blockName));
+       return Block.blockRegistry.getObject(new ResourceLocation(blockName));
     }
 
     static {
