--- original/net/minecraft/event/ClickEvent.java
+++ changed/net/minecraft/event/ClickEvent.java
@@ -37,14 +37,8 @@
                 return false;
             } else {
                 if (this.value != null) {
-                    if (!this.value.equals(clickevent.value)) {
-                        return false;
-                    }
-                } else if (clickevent.value != null) {
-                    return false;
-                }
-
-                return true;
+                    return this.value.equals(clickevent.value);
+                } else return clickevent.value == null;
             }
         } else {
             return false;
@@ -52,7 +46,7 @@
     }
 
     public String toString() {
-        return "ClickEvent{action=" + this.action + ", value=\'" + this.value + '\'' + '}';
+        return "ClickEvent{action=" + this.action + ", value='" + this.value + '\'' + '}';
     }
 
     public int hashCode() {
