--- original/net/minecraft/client/gui/GuiIngame.java
+++ changed/net/minecraft/client/gui/GuiIngame.java
@@ -6,6 +6,9 @@
 import java.util.Collection;
 import java.util.List;
 import java.util.Random;
+
+import dev.minecraftbox.event.EventBus;
+import dev.minecraftbox.event.events.rendering.RenderHUDEvent;
 import net.minecraft.block.material.Material;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.GlStateManager;
@@ -298,6 +301,7 @@
         this.mc.mcProfiler.endSection();
         GlStateManager.popMatrix();
         scoreobjective1 = scoreboard.getObjectiveInDisplaySlot(0);
+        EventBus.INSTANCE.post(new RenderHUDEvent(partialTicks));
 
         if (!this.mc.gameSettings.keyBindPlayerList.isKeyDown() || this.mc.isIntegratedServerRunning() && this.mc.thePlayer.sendQueue.getPlayerInfoMap().size() <= 1 && scoreobjective1 == null) {
             this.overlayPlayerList.updatePlayerList(false);
